common/src/main/scala/org/apache/celeborn/common/meta/ShufflePartitionLocationInfo.scala

```diff
@@ -65,6 +61,12 @@ class ShufflePartitionLocationInfo {
     removePartitions(slavePartitionLocations, partitionId)
   }
 
+  def removeAndGetAllMasterPartitionIds(): util.Set[Integer] = {
+    masterPartitionLocations.entrySet().asScala
```

Comment:
This method looks wired. Getting keySet and clearing the map will be more efficient.

Reply:
> This method looks wired. Getting keySet and clearing the map will be more efficient.

The semantics of this method means to delete all the current entries, so it's safe for new adding entries/replace element. Maybe we can simple replaced it by keyset.filter(remove(key)).toSet, but we need use that more carefully.